arrow/int.go

```diff
@@ -0,0 +1,105 @@
+package arrow
+
+import (
+	arrow "github.com/apache/arrow/go/arrow/array"
+	arrowmemory "github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/semantic"
+)
+
+type Int struct {
+	data *arrow.Int64
+}
+
+func (a *Int) Type() semantic.Type {
+	return semantic.Int
+}
+
+func (a *Int) IsNull(i int) bool {
+	return a.data.IsNull(i)
+}
+
+func (a *Int) IsValid(i int) bool {
+	return a.data.IsValid(i)
+}
+
+func (a *Int) Len() int {
+	return a.data.Len()
+}
+
+func (a *Int) NullN() int {
+	return a.data.NullN()
+}
+
+func (a *Int) Copy() *Int {
+	a.data.Retain()
+	return a
+}
+
+func (a *Int) Value(i int) int64 {
+	return a.data.Value(i)
+}
+
+func (a *Int) Int64Values() []int64 {
+	return a.data.Int64Values()
+}
+
+func (a *Int) Free() {
```

Comment:
Why are these named `Free` and `Copy` vs `Retain` and `Release`?

Reply:
I found `Retain` and `Release` personally confusing since I felt you had to then do the reference counting yourself. The original design had a wrapper called `BaseRef` that had the `Copy()` method but did not have `Free()` to signal that this wasn't a copy of the array that you  owned so, if you wanted to keep it, you needed to call `Copy()`. That got removed since it was part of the `array` interface.

I still like the overall idea though as `Copy` and `Free` are things that are easier to keep bug free in my opinion and I think it matches more closely with how we currently use it. The arrays once they are created are immutable so a "Copy" is free to be a shallow copy.

Reply:
@jsternberg the behavior of `Retain` and `Release` [are documented](https://github.com/apache/arrow/tree/master/go#reference-counting). Also worth noting is they are based off NSFoundation by Apple. Microsoft used a similar pattern, `AddRef` and `Release` for COM.

Reply:
We talked about this on Slack and I added that the reason I preferred the different API was so I could later add code that checked to make sure a single instance didn't free itself multiple times. I added that code to this PR so it is more obvious that a double free will not cause a hard-to-trace bug in the code as an added protection.

We may review this again in the future if the extra garbage that is generated causes a problem, but it should be minimal.

Reply:
As per my review, I think we should not modify the Go Arrow API and use Retain/Release directly.